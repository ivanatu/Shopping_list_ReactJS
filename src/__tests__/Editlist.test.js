import React from 'react';
import { shallow, render, mount } from 'enzyme';
import ReactDOM from 'react-dom' 
import renderer from 'react-test-renderer'
import Editlist from '../components/editlist'
import toJson from 'enzyme-to-json';
import '../setupTests';


describe('<Editlist/>', () => {
    const wrapper = shallow( <Editlist/> );

    it('renders without crashing', () => {
    render(<Editlist/>);
    expect(wrapper).toMatchSnapshot();
    });
    it('has a valid snapshot', () => {
        const component = renderer.create(
        <Editlist />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="modal"/>))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="modal-dialog"/>))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="modal-content"/>))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="modal-header"/>))
    });
    it('should render a button', () =>{
        expect(wrapper.find('button').length).toEqual(2)
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="modal-body"/>))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="form-group"/>))
    });
    it('input should respond to change event and change the state', () => {
        wrapper.find('#list').simulate('change', { target: { name: 'list', value: 'mangoes' } });
        expect(wrapper.state().name).toEqual('mangoes')
    });
    it('render the inputs', () =>{
        expect(wrapper.find('form').length).toEqual(1);
        expect(wrapper.find('input').length).toEqual(1)
    });
    it('renders the edit list form and submits data', () =>{
        wrapper.setState({list:'groceries'});
        wrapper.find("#editlist").simulate('submit', {preventDefault(){}});
        expect(toJson(wrapper)).toMatchSnapshot();
        // expect(handleSubmit.calledOnce).toEqual(false)
    });
    
});