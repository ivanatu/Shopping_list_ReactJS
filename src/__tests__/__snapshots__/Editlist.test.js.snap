// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Editlist/> has a valid snapshot 1`] = `
<div>
  <div
    className="modal"
    id="myModallundefined"
    role="dialog"
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h4>
            Edit List
          </h4>
          <button
            className="close"
            data-dismiss="modal"
            id="closeundefined"
            type="button"
          >
            ×
          </button>
        </div>
        <div
          className="modal-body"
        >
          <form
            className="register-form"
            id="editlist"
            method="POST"
            name="login_form"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              NEW LIST NAME:
              <input
                className="form-control"
                id="list"
                name="list"
                onChange={[Function]}
                required={true}
                type="text"
                value={undefined}
              />
            </div>
            <button
              className="btn btn-info "
              type="submit"
            >
              Save List
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Editlist/> renders the edit list form and submits data 1`] = `
<div>
  <div
    className="modal"
    id="myModallundefined"
    role="dialog"
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h4>
            Edit List
          </h4>
          <button
            className="close"
            data-dismiss="modal"
            id="closeundefined"
            type="button"
          >
            ×
          </button>
        </div>
        <div
          className="modal-body"
        >
          <form
            className="register-form"
            id="editlist"
            method="POST"
            name="login_form"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              NEW LIST NAME:
              <input
                className="form-control"
                id="list"
                name="list"
                onChange={[Function]}
                required={true}
                type="text"
                value="mangoes"
              />
            </div>
            <button
              className="btn btn-info "
              type="submit"
            >
              Save List
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Editlist/> renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="modal"
        id="myModallundefined"
        role="dialog"
      >
        <div
          className="modal-dialog"
        >
          <div
            className="modal-content"
          >
            <div
              className="modal-header"
            >
              <h4>
                Edit List
              </h4>
              <button
                className="close"
                data-dismiss="modal"
                id="closeundefined"
                type="button"
              >
                ×
              </button>
            </div>
            <div
              className="modal-body"
            >
              <form
                className="register-form"
                id="editlist"
                method="POST"
                name="login_form"
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  NEW LIST NAME:
                  <input
                    className="form-control"
                    id="list"
                    name="list"
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value={undefined}
                  />
                </div>
                <button
                  className="btn btn-info "
                  type="submit"
                >
                  Save List
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="modal-dialog"
        >
          <div
            className="modal-content"
          >
            <div
              className="modal-header"
            >
              <h4>
                Edit List
              </h4>
              <button
                className="close"
                data-dismiss="modal"
                id="closeundefined"
                type="button"
              >
                ×
              </button>
            </div>
            <div
              className="modal-body"
            >
              <form
                className="register-form"
                id="editlist"
                method="POST"
                name="login_form"
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  NEW LIST NAME:
                  <input
                    className="form-control"
                    id="list"
                    name="list"
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value={undefined}
                  />
                </div>
                <button
                  className="btn btn-info "
                  type="submit"
                >
                  Save List
                </button>
              </form>
            </div>
          </div>
        </div>,
        "className": "modal",
        "id": "myModallundefined",
        "role": "dialog",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="modal-content"
          >
            <div
              className="modal-header"
            >
              <h4>
                Edit List
              </h4>
              <button
                className="close"
                data-dismiss="modal"
                id="closeundefined"
                type="button"
              >
                ×
              </button>
            </div>
            <div
              className="modal-body"
            >
              <form
                className="register-form"
                id="editlist"
                method="POST"
                name="login_form"
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  NEW LIST NAME:
                  <input
                    className="form-control"
                    id="list"
                    name="list"
                    onChange={[Function]}
                    required={true}
                    type="text"
                    value={undefined}
                  />
                </div>
                <button
                  className="btn btn-info "
                  type="submit"
                >
                  Save List
                </button>
              </form>
            </div>
          </div>,
          "className": "modal-dialog",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="modal-header"
              >
                <h4>
                  Edit List
                </h4>
                <button
                  className="close"
                  data-dismiss="modal"
                  id="closeundefined"
                  type="button"
                >
                  ×
                </button>
              </div>,
              <div
                className="modal-body"
              >
                <form
                  className="register-form"
                  id="editlist"
                  method="POST"
                  name="login_form"
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    NEW LIST NAME:
                    <input
                      className="form-control"
                      id="list"
                      name="list"
                      onChange={[Function]}
                      required={true}
                      type="text"
                      value={undefined}
                    />
                  </div>
                  <button
                    className="btn btn-info "
                    type="submit"
                  >
                    Save List
                  </button>
                </form>
              </div>,
            ],
            "className": "modal-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    Edit List
                  </h4>,
                  <button
                    className="close"
                    data-dismiss="modal"
                    id="closeundefined"
                    type="button"
                  >
                    ×
                  </button>,
                ],
                "className": "modal-header",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Edit List",
                  },
                  "ref": null,
                  "rendered": "Edit List",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "×",
                    "className": "close",
                    "data-dismiss": "modal",
                    "id": "closeundefined",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "×",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <form
                  className="register-form"
                  id="editlist"
                  method="POST"
                  name="login_form"
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    NEW LIST NAME:
                    <input
                      className="form-control"
                      id="list"
                      name="list"
                      onChange={[Function]}
                      required={true}
                      type="text"
                      value={undefined}
                    />
                  </div>
                  <button
                    className="btn btn-info "
                    type="submit"
                  >
                    Save List
                  </button>
                </form>,
                "className": "modal-body",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="form-group"
                    >
                      NEW LIST NAME:
                      <input
                        className="form-control"
                        id="list"
                        name="list"
                        onChange={[Function]}
                        required={true}
                        type="text"
                        value={undefined}
                      />
                    </div>,
                    <button
                      className="btn btn-info "
                      type="submit"
                    >
                      Save List
                    </button>,
                  ],
                  "className": "register-form",
                  "id": "editlist",
                  "method": "POST",
                  "name": "login_form",
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "NEW LIST NAME:",
                        <input
                          className="form-control"
                          id="list"
                          name="list"
                          onChange={[Function]}
                          required={true}
                          type="text"
                          value={undefined}
                        />,
                      ],
                      "className": "form-group",
                    },
                    "ref": null,
                    "rendered": Array [
                      "NEW LIST NAME:",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "form-control",
                          "id": "list",
                          "name": "list",
                          "onChange": [Function],
                          "required": true,
                          "type": "text",
                          "value": undefined,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Save List",
                      "className": "btn btn-info ",
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "Save List",
                    "type": "button",
                  },
                ],
                "type": "form",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="modal"
          id="myModallundefined"
          role="dialog"
        >
          <div
            className="modal-dialog"
          >
            <div
              className="modal-content"
            >
              <div
                className="modal-header"
              >
                <h4>
                  Edit List
                </h4>
                <button
                  className="close"
                  data-dismiss="modal"
                  id="closeundefined"
                  type="button"
                >
                  ×
                </button>
              </div>
              <div
                className="modal-body"
              >
                <form
                  className="register-form"
                  id="editlist"
                  method="POST"
                  name="login_form"
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    NEW LIST NAME:
                    <input
                      className="form-control"
                      id="list"
                      name="list"
                      onChange={[Function]}
                      required={true}
                      type="text"
                      value={undefined}
                    />
                  </div>
                  <button
                    className="btn btn-info "
                    type="submit"
                  >
                    Save List
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="modal-dialog"
          >
            <div
              className="modal-content"
            >
              <div
                className="modal-header"
              >
                <h4>
                  Edit List
                </h4>
                <button
                  className="close"
                  data-dismiss="modal"
                  id="closeundefined"
                  type="button"
                >
                  ×
                </button>
              </div>
              <div
                className="modal-body"
              >
                <form
                  className="register-form"
                  id="editlist"
                  method="POST"
                  name="login_form"
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    NEW LIST NAME:
                    <input
                      className="form-control"
                      id="list"
                      name="list"
                      onChange={[Function]}
                      required={true}
                      type="text"
                      value={undefined}
                    />
                  </div>
                  <button
                    className="btn btn-info "
                    type="submit"
                  >
                    Save List
                  </button>
                </form>
              </div>
            </div>
          </div>,
          "className": "modal",
          "id": "myModallundefined",
          "role": "dialog",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="modal-content"
            >
              <div
                className="modal-header"
              >
                <h4>
                  Edit List
                </h4>
                <button
                  className="close"
                  data-dismiss="modal"
                  id="closeundefined"
                  type="button"
                >
                  ×
                </button>
              </div>
              <div
                className="modal-body"
              >
                <form
                  className="register-form"
                  id="editlist"
                  method="POST"
                  name="login_form"
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    NEW LIST NAME:
                    <input
                      className="form-control"
                      id="list"
                      name="list"
                      onChange={[Function]}
                      required={true}
                      type="text"
                      value={undefined}
                    />
                  </div>
                  <button
                    className="btn btn-info "
                    type="submit"
                  >
                    Save List
                  </button>
                </form>
              </div>
            </div>,
            "className": "modal-dialog",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="modal-header"
                >
                  <h4>
                    Edit List
                  </h4>
                  <button
                    className="close"
                    data-dismiss="modal"
                    id="closeundefined"
                    type="button"
                  >
                    ×
                  </button>
                </div>,
                <div
                  className="modal-body"
                >
                  <form
                    className="register-form"
                    id="editlist"
                    method="POST"
                    name="login_form"
                    onSubmit={[Function]}
                  >
                    <div
                      className="form-group"
                    >
                      NEW LIST NAME:
                      <input
                        className="form-control"
                        id="list"
                        name="list"
                        onChange={[Function]}
                        required={true}
                        type="text"
                        value={undefined}
                      />
                    </div>
                    <button
                      className="btn btn-info "
                      type="submit"
                    >
                      Save List
                    </button>
                  </form>
                </div>,
              ],
              "className": "modal-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h4>
                      Edit List
                    </h4>,
                    <button
                      className="close"
                      data-dismiss="modal"
                      id="closeundefined"
                      type="button"
                    >
                      ×
                    </button>,
                  ],
                  "className": "modal-header",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Edit List",
                    },
                    "ref": null,
                    "rendered": "Edit List",
                    "type": "h4",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "×",
                      "className": "close",
                      "data-dismiss": "modal",
                      "id": "closeundefined",
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "×",
                    "type": "button",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <form
                    className="register-form"
                    id="editlist"
                    method="POST"
                    name="login_form"
                    onSubmit={[Function]}
                  >
                    <div
                      className="form-group"
                    >
                      NEW LIST NAME:
                      <input
                        className="form-control"
                        id="list"
                        name="list"
                        onChange={[Function]}
                        required={true}
                        type="text"
                        value={undefined}
                      />
                    </div>
                    <button
                      className="btn btn-info "
                      type="submit"
                    >
                      Save List
                    </button>
                  </form>,
                  "className": "modal-body",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="form-group"
                      >
                        NEW LIST NAME:
                        <input
                          className="form-control"
                          id="list"
                          name="list"
                          onChange={[Function]}
                          required={true}
                          type="text"
                          value={undefined}
                        />
                      </div>,
                      <button
                        className="btn btn-info "
                        type="submit"
                      >
                        Save List
                      </button>,
                    ],
                    "className": "register-form",
                    "id": "editlist",
                    "method": "POST",
                    "name": "login_form",
                    "onSubmit": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "NEW LIST NAME:",
                          <input
                            className="form-control"
                            id="list"
                            name="list"
                            onChange={[Function]}
                            required={true}
                            type="text"
                            value={undefined}
                          />,
                        ],
                        "className": "form-group",
                      },
                      "ref": null,
                      "rendered": Array [
                        "NEW LIST NAME:",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "className": "form-control",
                            "id": "list",
                            "name": "list",
                            "onChange": [Function],
                            "required": true,
                            "type": "text",
                            "value": undefined,
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "input",
                        },
                      ],
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Save List",
                        "className": "btn btn-info ",
                        "type": "submit",
                      },
                      "ref": null,
                      "rendered": "Save List",
                      "type": "button",
                    },
                  ],
                  "type": "form",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
